mlrun/runtimes/base.py

```diff
@@ -1121,10 +1170,84 @@ def _is_runtime_resource_run_in_transient_state(
 
         return False, last_update
 
+    def _verify_run_reached_stable_state_and_monitor_progress(
+        self, db: DBInterface, db_session: Session, project: str, run_uid: str
+    ):
+        """
+        This function runs in a retry_until_successful and therefore need to raise if run didn't reach stable state
+        """
+        desired_run_state = self._resolve_run_state_representing_current_resource_state(
```

Comment:
Maybe this is better cleared offline but you're using the word "desired" here on the object that represents the REAL state.
Usually desired_state (== target_state) is a different concept.
This is also true inside the helper functions `desired_run_state` is used in a way that I think is not accurate

Reply:
Explained above the idea of this function
> This function is actually checking the runtime resource, resolving its state, and translating it to what should be the representing run state
I agree both function and variable naming are not great, I accept suggestions if you have something

Reply:
This function does not exist anymore (as part of moving to one loop to monitor all runs)
but `desired_run_state` still exists and is the output of either `_resolve_pod_status_info` or `_resolve_crd_object_status_info`